examples/apps/cli/cli_uart.cpp

```diff
@@ -147,15 +147,12 @@ static void ReceiveTask(const uint8_t *aBuf, uint16_t aBufLength)
         {
         case '\r':
         case '\n':
-            Output(CRNL, sizeof(CRNL));
-            if (sRxLength > 0)
+            if (!(sLastChar == '\r' && *aBuf == '\n'))
```

Comment:
How about moving this check under `case '\n'`?  I guess the goal is to skip if last is `\r` and new char is `\n`?

- Under `case \n`, we can do ` if (sLastChar == '\r')` and if it matches we `break`, otherwise fall through to `case \r`?